titan-compiler/ast.lua

```diff
@@ -54,21 +54,32 @@ types.Args = {
 
 -- Create a function for each type constructor
 for typename, conss in pairs(types) do
-    ast[typename] = {}
     for consname, fields in pairs(conss) do
-        ast[typename][consname] = function(...)
+        local tag = typename .. '_' .. consname
+
+        local mt = { __index = {
+            traverse = function(self, visitor, ...)
```

Comment:
This version of traverse is not recursive. Looks more like a foreach / map than a traverse...

Reply:
Im going to rename it to `foreach`.